src/main/resources/com/google/api/codegen/py/setup.py.snip

```diff
@@ -12,10 +12,16 @@
   import sys
 
   install_requires = [
-      'googleapis-common-protos[grpc]>={@metadata.commonProtosVersionBound.lower}, <{@metadata.commonProtosVersionBound.upper}',
       'google-gax>={@metadata.gaxVersionBound.lower}, <{@metadata.gaxVersionBound.upper}',
-      '{@metadata.protoPackageName}[grpc]>={@metadata.packageVersionBound.lower}, <{@metadata.packageVersionBound.upper}',
       'oauth2client>={@metadata.authVersionBound.lower}, <{@metadata.authVersionBound.upper}',
+      '{@metadata.protoPackageName}[grpc]>={@metadata.packageVersionBound.lower}, <{@metadata.packageVersionBound.upper}',
+      @join packageDep : metadata.protoPackageDependencies
+        @if or(packageDep.name == "googleapis-common-protos", packageDep.name.startsWith("proto-"))
+          '{@packageDep.name}[grpc]>={@packageDep.versionBound.lower}, <{@packageDep.versionBound.upper}',
+        @else
```

Comment:
The test case was `python-package`, and I was concerned that there would be a case where someone did not use it how you intended. Essentially, I do not want to get an error by referencing a `[grpc]` on a package that does not have one.